app/src/test/java/org/mozilla/fenix/settings/advanced/LocaleSettingsStoreTest.kt

```diff
@@ -42,4 +42,12 @@ class LocaleSettingsStoreTest {
         assertEquals(2, (localeSettingsStore.state.searchedLocaleList as ArrayList).size)
         assertEquals(selectedLocale, localeSettingsStore.state.searchedLocaleList[1])
     }
+
+    @Test
+    fun `reset search list upon selection`() = runTest {
```

Comment:
We are following a "GIVEN smth WHEN smth THEN smth" naming pattern for tests lately.

Since we are interested in making sure that the locale list is reset after selecting a locale, checking its state before selection as well as after sounds reasonable.